web/app/js/components/util/ApiHelpers.jsx

```diff
@@ -1,12 +1,34 @@
+import _ from 'lodash';
+import { Link } from 'react-router-dom';
+import React from 'react';
 import 'whatwg-fetch';
 
-export const ApiHelpers = pathPrefix => {
-  const podsPath = `${pathPrefix}/api/pods`;
+export const ApiHelpers = (pathPrefix, defaultMetricsWindow = '10m') => {
+  let metricsWindow = defaultMetricsWindow;
+  const podsPath = `/api/pods`;
+  const metricsPath = () => `/api/metrics?window=${metricsWindow}`;
+
+  const validMetricsWindows = {
+    "10s": true,
+    "1m": true,
+    "10m": true,
+    "1h": true
+  };
 
   const apiFetch = path => {
+    if (!_.isEmpty(pathPrefix)) {
+      path = `${pathPrefix}${path}`;
+    }
     return fetch(path).then(handleFetchErr).then(r => r.json());
   };
 
+  const fetchMetrics = (path = metricsPath()) => {
+    if (path.indexOf("window") === -1) {
+      path = `${path}&window=${getMetricsWindow()}`;
```

Comment:
will this generate URLs like `/foo/bar&window=baz` ? should we ensure we're inserting a `?` between the path and query string?

Reply:
good call!